### ISSUE TITLE
Diff between this project and JCodeModel

### ISSUE THREAD

**Issue by leventov** [square/javapoet#221]
How JavaPoet is compared to JCodeModel (see github.com/phax/jcodemodel)? Has the latter been considered when you initially had Java code generation at Square? When one or another project should be used?

**Comment by swankjesse** [comment]
The JavaPoet docs are pretty clear about what it does and how to get started. I haven't used JCodeModel. (I also didn't see the getting started docs. I'd like to see a guide like our readme on how to generate various things!)

**Comment by leventov** [comment]
Your can start with http://phax.github.io/jcodemodel/site/apidocs/com/helger/jcodemodel/JCodeModel.html and http://blogtech.cardosi.net/2013/03/19/tutorial-java-codemodel-basics/

Now your might think that suggestion to compare a project with 5 stars with a project with 700 is a joke,  but JCodeModel is old project, formerly maintained by Sun, when you not even started JavaPoet or it's predecessors, it has already been here. That's why I wondered if you considered this before writing own wheel.

**Comment by swankjesse** [comment]
I didn't adopt JCodeModel is that I didn't know it existed. JavaPoet development started in 2012; JCodeModel wasn't on GitHub until 2013. Perhaps you should ask the JCodeModel maintainers the same question!

I'll continue to avoid JCodeModel is that it has a GPL license. I do not use GPL-licensed code in my projects because I am not prepared to satisfy the licensing requirements. I want to redistribute JavaPoet in commercial, closed source applications and the GPL doesn't permit that. (I was named in the Oracle-Google trial, so I take this very seriously.)

**Comment by swankjesse** [comment]
By the way, those tutorials are handy. You should link 'em from the JCodeModel GitHub page. Or even better, include 'em directly.

**Comment by leventov** [comment]
Seems that JCodeModel started in 2005: https://codemodel.java.net/ Presence on Github wouldn't help you much, you anyway need to search for projects via Google, somehow. For example, when I faced code generation problem in Java I found JCodeModel, but haven't found JavaPoet.

Agreed, GPL is valid reason for starting new project. Moreover, JCodeModel doesn't seem to be a project of excellent quality and comprehensiveness; it is merely solves the problem, but doesn't shine.
